src/opcode.h

```diff
@@ -0,0 +1,284 @@
+  V(F32, F32, F32, 0, 0x97, F32Max, "f32.max")                        \
+  V(F32, F32, F32, 0, 0x98, F32Copysign, "f32.copysign")              \
+  V(F64, F64, F64, 0, 0x99, F64Abs, "f64.abs")                        \
+  V(F64, F64, F64, 0, 0x9a, F64Neg, "f64.neg")                        \
+  V(F64, F64, F64, 0, 0x9b, F64Ceil, "f64.ceil")                      \
+  V(F64, F64, F64, 0, 0x9c, F64Floor, "f64.floor")                    \
+  V(F64, F64, F64, 0, 0x9d, F64Trunc, "f64.trunc")                    \
+  V(F64, F64, F64, 0, 0x9e, F64Nearest, "f64.nearest")                \
+  V(F64, F64, F64, 0, 0x9f, F64Sqrt, "f64.sqrt")                      \
+  V(F64, F64, F64, 0, 0xa0, F64Add, "f64.add")                        \
+  V(F64, F64, F64, 0, 0xa1, F64Sub, "f64.sub")                        \
+  V(F64, F64, F64, 0, 0xa2, F64Mul, "f64.mul")                        \
+  V(F64, F64, F64, 0, 0xa3, F64Div, "f64.div")                        \
+  V(F64, F64, F64, 0, 0xa4, F64Min, "f64.min")                        \
+  V(F64, F64, F64, 0, 0xa5, F64Max, "f64.max")                        \
+  V(F64, F64, F64, 0, 0xa6, F64Copysign, "f64.copysign")              \
+  V(I32, I64, ___, 0, 0xa7, I32WrapI64, "i32.wrap/i64")               \
+  V(I32, F32, ___, 0, 0xa8, I32TruncSF32, "i32.trunc_s/f32")          \
+  V(I32, F32, ___, 0, 0xa9, I32TruncUF32, "i32.trunc_u/f32")          \
+  V(I32, F64, ___, 0, 0xaa, I32TruncSF64, "i32.trunc_s/f64")          \
+  V(I32, F64, ___, 0, 0xab, I32TruncUF64, "i32.trunc_u/f64")          \
+  V(I64, I32, ___, 0, 0xac, I64ExtendSI32, "i64.extend_s/i32")        \
+  V(I64, I32, ___, 0, 0xad, I64ExtendUI32, "i64.extend_u/i32")        \
+  V(I64, F32, ___, 0, 0xae, I64TruncSF32, "i64.trunc_s/f32")          \
+  V(I64, F32, ___, 0, 0xaf, I64TruncUF32, "i64.trunc_u/f32")          \
+  V(I64, F64, ___, 0, 0xb0, I64TruncSF64, "i64.trunc_s/f64")          \
+  V(I64, F64, ___, 0, 0xb1, I64TruncUF64, "i64.trunc_u/f64")          \
+  V(F32, I32, ___, 0, 0xb2, F32ConvertSI32, "f32.convert_s/i32")      \
+  V(F32, I32, ___, 0, 0xb3, F32ConvertUI32, "f32.convert_u/i32")      \
+  V(F32, I64, ___, 0, 0xb4, F32ConvertSI64, "f32.convert_s/i64")      \
+  V(F32, I64, ___, 0, 0xb5, F32ConvertUI64, "f32.convert_u/i64")      \
+  V(F32, F64, ___, 0, 0xb6, F32DemoteF64, "f32.demote/f64")           \
+  V(F64, I32, ___, 0, 0xb7, F64ConvertSI32, "f64.convert_s/i32")      \
+  V(F64, I32, ___, 0, 0xb8, F64ConvertUI32, "f64.convert_u/i32")      \
+  V(F64, I64, ___, 0, 0xb9, F64ConvertSI64, "f64.convert_s/i64")      \
+  V(F64, I64, ___, 0, 0xba, F64ConvertUI64, "f64.convert_u/i64")      \
+  V(F64, F32, ___, 0, 0xbb, F64PromoteF32, "f64.promote/f32")         \
+  V(I32, F32, ___, 0, 0xbc, I32ReinterpretF32, "i32.reinterpret/f32") \
+  V(I64, F64, ___, 0, 0xbd, I64ReinterpretF64, "i64.reinterpret/f64") \
+  V(F32, I32, ___, 0, 0xbe, F32ReinterpretI32, "f32.reinterpret/i32") \
+  V(F64, I64, ___, 0, 0xbf, F64ReinterpretI64, "f64.reinterpret/i64")
```

Comment:
What do you think about putting this list in something like "opcode.def" and then including it each time you need it?  I quite like this pattern as it keeps long data listings separate from source.